### ISSUE TITLE
`SweptTube`: name the one-parameter family of slices, once it earns a type

### ISSUE THREAD

**Issue by nstarman** [GalacticDynamics/coordinax#829]
Deferred from #828. Recording the shape and the argument so the decision is not re-derived.

## The object

The tube swept through time — the 4-D thing whose two 3-D sections #814 separated: the **worldtube** (station pinned, coordinates $(t, n_1, n_2)$) and the **spatial slice** (time pinned by `AtTime`, coordinates $(\sigma, n_1, n_2)$).

Right now it has no representation. It exists only as a lambda that callers write out:

```python
K = cxfc.rate_of_strain(
    lambda t: cxfc.TubularChart(
        cxfc.BishopBuilder(cxfc.AtTime(curve, t), "km"), tau_bounds=bounds
    ),
    point, u.Q(1.0, "s"),
)
```

A type would own the recipe — curve, `tau_unit`, `tau_bounds`, builder class — and yield charts:

```python
tube = cxfc.SweptTube(curve, "km", tau_bounds=bounds)

tube.at(u.Q(1.0, "s"))                     # the spatial slice: a TubularChart
tube.worldtube(u.Q(1.3, "km"))             # the other section
tube.rate_of_strain(point, u.Q(1.0, "s"))
```

## On the name

`SliceFamily` was the working name and is accurate but says nothing about what.

`TubularFoliation` / `TubularSlicing` borrow precise 3+1 vocabulary and name the wrong half of it: a *slicing* is the choice of $t=\text{const}$ surfaces — the lapse — whereas the material-vs-Eulerian choice this object carries is the *threading*, how spatial coordinates connect between slices. Committing to a term that is subtly wrong is worse than plain English.

`worldtube` is already established in the curveframes docs (16 uses) for the station-pinned section, so it is not available for the whole.

**`SweptTube`** names the 4-D object, reads at both call sites, and needs no jargon that could be misapplied.

## Why it is not built yet

One method does not earn a type, and the argument I expected to carry it did not survive being checked. The docs warn that `AtTime`/`ArcLength` ordering changes what is being asked, so hand-rolling the family looked error-prone in a way a type would prevent. Measured on $K_{\tau\tau}$, both orderings give `-0.005485` — identical. The footgun is not there for this quantity.

Against building it now:

- exactly one method today
- it re-states the chart recipe, so `(curve, tau_unit, tau_bounds)` would live in two places unless derived from an existing builder — and two sources of truth for one fact caused a good share of the bugs #773, #805, #814 and #820 fixed

## What would earn it

- a **second** time-dependent quantity wanting the same family
- the `.at()` / `.worldtube()` symmetry becoming load-bearing, so both sections derive from one object rather than being assembled separately
- a measured case where hand-rolling the family *does* silently give a wrong answer

Any of those, and it can wrap `rate_of_strain` without breaking a caller.

**Comment by nstarman** [comment]
## What #870 changes about this issue

This issue asks what would *earn* `SweptTube`. #870 supplies the third bullet outright — "a measured case where hand-rolling the family does silently give a wrong answer" — and strengthens it: the family cannot be hand-rolled correctly **at all** without director data the curve does not contain. The type is no longer a convenience wrapper; it is where the gauge gets a home.

Porting the design from the unproposed branch `claude/swepttube-gauge-spec`, which is being deleted rather than committed as a tracked spec file. The evidence for *why no seed rule works* went to #870.

## Design

```python
class SweptTube(eqx.Module):
    """One-parameter family of tubular slices: t -> TubularChart."""

    curve: Any                                    # two-argument gamma(tau, t)
    director: Callable[[Any], Any] | None = None  # required iff builder is Bishop
    tau_unit: ... = eqx.field(static=True)
    tau_bounds: tuple[Any, Any]                   # kw_only
    builder: type = eqx.field(static=True, default=BishopBuilder)
    # solver / n_seed passthrough

    def __call__(self, t) -> TubularChart: ...
```

`director` is a **callable of $t$**, not a fixed vector, because a materially spinning rod needs a $t$-dependent frame. It is required **only on the Bishop path**: `FrenetSerretBuilder` takes no seed, so passing one is a caller error rejected at construction rather than silently ignored. (An earlier draft said "omitting it raises" *and* offered a selectable builder — those two rules contradict each other.)

`builder` carries the choice because `FrenetSerretBuilder` needs no seed and is **already** equivariant ($7.6\times10^{-17}$, exactly `0.0` on axis-aligned rotations) — $\mathbf{N}$ and $\mathbf{B}$ are fixed pointwise by the curve. The gauge bug is Bishop-specific. Frenet is also required for the off-diagonal acceptance test.

### `rate_of_strain` narrows to `SweptTube`

The raw-callable overload is removed. Note what that does and does **not** buy: `director` is itself an opaque `Callable`, so the gauge is no more *inspectable* than before. Narrowing buys exactly one thing — **omission raises** rather than silently defaulting (criterion 5). An earlier draft's claim that it makes the contract "structural rather than advisory" overstated it.

It must additionally either invoke the reach guard or document that it is unvalidated outside the tube: at present, past the focal distance $\gamma_{\tau\tau} = 1.379\times10^{-3}$ with nothing raising, and the trace identity breaks by two orders *at* the focal distance ($8.6\times10^{-2}$ vs $7.736$), because `metric_matrix` / `rate_of_strain` never call `TubularChart.check_data(values=True)`.

> ### Day-1 blocker, found by building the sketch
> A `Quantity`-valued director — the natural typing, and what the rest of this library uses — raises `TracerArrayConversionError` inside `rate_of_strain`'s `jacfwd`. `_float` in `bishop.py` calls `jnp.asarray(x)`, which invokes `__array__` on a traced `Quantity`, reached via `_transport_start`'s `_orthonormalize(_float(self.initial_normal), T0_val)`. A raw `jnp` array works. One line to fix (`ustrip` first), but it must land before or with this work. **Re-anchor the line number against current `main`.**

## Acceptance criteria

The first draft had eight. **Four had provably zero power** over the gauge (#3 was on a straight line where $\kappa = 0$; #4 was an algebraic identity true for any $\gamma(t)$ and passes on the broken implementation; #5's symmetry half is vacuous for both builders; #8 was at $n = 0$ where the gauge cancels). The proposed implementation passed all eight and was still wrong. Revised:

| # | Check | Power |
| --- | --- | --- |
| 1 | Rigid rotation, **generic axis**, materially correct director: $\lvert K\rvert < 10^{-9}$ | The headline |
| 2 | Same rotation, constant director $\perp \mathbf{T}(\tau_0)$: assert $K_{\tau\tau}$ against $\lvert r'\rvert^2(1-n\cdot k)\,a\,(n\times k)_z$ with $a = \boldsymbol\omega\cdot\mathbf{T}(\tau_0)$. **Both conditions load-bearing**: $\mathbf{T}(\tau)$ instead of $\mathbf{T}(\tau_0)$ gives 0.814815 against 0.928477 (14% error), and a non-perpendicular director breaks the identity (0.104803 vs 0.108417). State the configuration alongside the number | Pins that the caller's gauge is reported faithfully. **Not** "$\neq 0$", which passes on 1e-14 solver noise |
| 3 | Breathing circle $R(t) = 1 + t/2$ **in the angle parametrisation** $r = R(t)(\cos\tau,\sin\tau)$: $K_{\tau\tau} = (R + n_1)\dot R$ at $n_1 = 0,\pm0.3$ — measured 0.500000 / 0.650000 / 0.350000, director the outward radial at $\tau_0$. The formula is **wrong under arc length**, where the same criterion yields 0.000000 / −0.195 / +0.105 | Analytic, but **narrower than it looks**: $T(\tau_0)$ does not rotate in $t$, so the status-quo auto seed is constant and passes. Discriminates against a rotated director, not against the seed-drift bug |
| 4 | Frenet off-diagonals against closed form $\gamma_{\tau n_1} = -\lVert\gamma'\rVert\sigma n_2$, $\gamma_{\tau n_2} = +\lVert\gamma'\rVert\sigma n_1$ | Reference values; "$\neq 0$" passes on a sign error |
| 5 | Missing `director` raises **on the Bishop path**; supplying one with `builder=FrenetSerretBuilder` also raises | The core contract. A dataclass field cannot be "required iff builder is Bishop", so `SweptTube` needs its own conditional check — `FrenetSerretBuilder` rejects `initial_normal` with a `TypeError` one layer down, a different error arriving too late |
| 6 | `jit` and `vmap` over scalar `t` | Killed the earlier guard rule; see #872, where a Python `float()` on the offset blocked every tracer |
| 7 | Past the focal distance it **raises** — the "or documented unvalidated" escape is deleted, since an implementation passed either way. Assert the singular set **per builder**: Bishop $1-k_1n_1-k_2n_2=0$, Frenet $1-\kappa n_1=0$ (independent of $n_2$ — `n=(1.5,0)` and `n=(1.5,0.7)` both give `-0.29310345`) | D-series gap |
| 8 | $K_{n_in_j} \equiv 0$ pinned as a known structural limit | Documents rather than tests |
| 9 | Uniform stretch of a straight line, $K_{\tau\tau} = c(1+ct)$ | Plumbing only — no gauge power; keep, labelled as such |
| 10 | $\gamma^{ij}K_{ij} = \partial_t\ln\sqrt{\det\gamma}$ | Plumbing only — labelled as such |
| 11 | **Static curve, spinning director.** Helix $(\cos s,\sin s,0.4s)$ with no $t$-dependence, `tau_bounds` $=(-1,2)$ km, at $(\tau,n_1,n_2)=(0.5,0.2,0.1)$ km, director spun about $\mathbf{T}(\tau_0)$ at $a = 1/\sqrt{1.16}$: assert $K_{\tau\tau} = -0.067526$ against **exactly 0.0** for a fixed director | **The criterion the design exists for.** No curve-derived rule can pass it — the curve is identical in both cases. Without it the design rests on criterion 1 alone, a zero-truth $\lvert K\rvert < 10^{-9}$ check |

> **A verification that lands where the truth is zero proves nothing.** The first audit's `_tau_of_s` $t$-direction check sat at $t=2.0$, where the true derivative is analytically exactly zero; AD and finite differences agreed to 3.4e-11 and confirmed nothing. Re-run at $t=1.0$ and $t=3.0$ it genuinely matches (7.1e-12, 1.4e-11). Every criterion must be checked for this — the same defect made four of the original eight vacuous.

## Sequencing

`BishopBuilder` requiring an explicit seed is **independent of this issue** and must not ride in the same PR: that PR would be unreviewable and unrevertable.

1. **`BishopBuilder` breaking change, on its own.** `_auto_initial_normal` removed; no-seed construction raises. By the #870 lemma a chart's $(n_1,n_2)$ genuinely depend on the seed whenever $\kappa \neq 0$ and $n \neq 0$, so a caller who never chose it still depends on it. **Scale, counted not estimated: 156 `BishopBuilder(` construction sites across 28 files, 11 of which already pass `initial_normal` — so 145 need a seed.** (Earlier drafts got this wrong twice: "~258 references across 25+ files" counted references not constructions; "157 / 38 / 8" came from a grep that counted the spec file itself.)
   > The *stated justification* was also wrong in an earlier draft, which called the auto seed "arbitrary and undocumented". Arbitrary yes; undocumented no — it is described in `bishop.py` and in `_auto_initial_normal`'s own docstring. The honest case is that the gauge is arbitrary **and load-bearing**.
2. **`SweptTube`.** The type, the required director, criteria 1–11. Closes this issue.
3. **Documentation.** $K_{ij} = \tfrac12(\mathcal{L}_T\gamma)_{ij}$ with $T$ the tube's declared time flow. Say **"by analogy with ADM"**, not "the lapse is 1": Galilean spacetime has no non-degenerate 4-metric, so lapse and shift are not defined. State that $K$ is a slice 2-tensor only under *time-independent* spatial relabelling, that rate of strain of a tube is a property of a framed curve, and that $K_{n_in_j} \equiv 0$.

## Behaviour regressions to declare

1. `BishopBuilder` without `initial_normal` raises (145 construction sites).
2. `rate_of_strain` no longer accepts a raw callable.
3. `SweptTube` requires a `director`; there is no automatic gauge.

## Sign convention — stated, and **not yet verified**

$K_{ij} = +\tfrac{1}{2\alpha}(\partial_t\gamma_{ij} - \mathcal{L}_\beta\gamma_{ij})$ is Wald's sign ($K_{ab} = \nabla_a n_b$). Baumgarte–Shapiro (2.134) and Alcubierre (2.3.11) are believed to write $\partial_t\gamma_{ij} = -2\alpha K_{ij} + D_i\beta_j + D_j\beta_i$, the opposite sign, following MTW. For a function named `rate_of_strain` the positive sign is right, but the convention must be stated or anyone assembling evolution equations from a numerical-relativity text inherits a sign error.

> **This attribution was made from memory of the standard texts and has never been checked against the books.** Confirm it against the actual editions before it is written into the docs.
